elements/pl-order-blocks/pl-order-blocks.js

```diff
@@ -1,112 +1,90 @@
+    leftDiff = Math.round(leftDiff / TABWIDTH) * TABWIDTH;
+    let currentIndent = ui.item[0].style.marginLeft;
+    if (currentIndent !== '') {
+      leftDiff += parseInt(currentIndent);
+    }
 
-  // limit leftDiff to be in within the bounds of the drag and drop box
-  // that is, at least indented 0 times, or at most indented by MAX_INDENT times
-  leftDiff = Math.min(leftDiff, TABWIDTH * MAX_INDENT);
-  leftDiff = Math.max(leftDiff, 0);
+    // limit leftDiff to be in within the bounds of the drag and drop box
+    // that is, at least indented 0 times, or at most indented by MAX_INDENT times
+    leftDiff = Math.min(leftDiff, TABWIDTH * maxIndent);
+    leftDiff = Math.max(leftDiff, 0);
 
-  return leftDiff;
-}
+    return leftDiff;
+  }
 
-$(document).ready(function () {
-  // Add drag and drop functionality for options elements that
-  // has the pl-order-blocks-connected-sortable class code for the HTML popover
-  $('.pl-order-blocks-connected-sortable').sortable({
-    items: 'li:not(.info-fixed)',
-    cancel: '.info',
-    connectWith: '.pl-order-blocks-connected-sortable',
-    placeholder: 'ui-state-highlight',
-    create: function (event) {
-      // when the sortable is created, we need to put the two functions here
-      // to restore progress when the user refresh/submits an answer
-      set_answer(event);
-      set_max_indent(event);
-    },
-    sort: function (event, ui) {
-      // update the location of the placeholder as the item is dragged
-      let placeholder = ui.placeholder;
-      let leftDiff = calculate_indent(ui, placeholder.parent());
-      placeholder[0].style.marginLeft = leftDiff + 'px';
-    },
-    beforeStop: function (event, ui) {
-      if (!check_block(event, ui)) {
-        $(this).sortable('cancel');
-      }
-    },
-    stop: function (event, ui) {
-      // when the user stops interacting with the list
-      let leftDiff = calculate_indent(ui, ui.item.parent());
-      ui.item[0].style.marginLeft = leftDiff + 'px';
-      set_answer(event);
-    },
-  });
+  function initialize(elementId, connectWith) {
+    $(elementId).sortable({
+      items: 'li:not(.info-fixed)',
+      cancel: '.info',
+      connectWith: connectWith,
+      placeholder: 'ui-state-highlight',
+      create: function (event) {
+        setAnswer(event);
+      },
+      sort: function (event, ui) {
+        // update the location of the placeholder as the item is dragged
+        let placeholder = ui.placeholder;
+        let leftDiff = calculateIndent(ui, placeholder.parent());
+        placeholder[0].style.marginLeft = leftDiff + 'px';
+      },
+      stop: function (event, ui) {
+        // when the user stops interacting with the list
+        let leftDiff = calculateIndent(ui, ui.item.parent());
+        ui.item[0].style.marginLeft = leftDiff + 'px';
+        setAnswer(event);
+      },
+    });
+  }
+
+  initialize(optionsElementId, dropzoneElementId);
+  initialize(dropzoneElementId, optionsElementId);
```

Comment:
Is it necessary to `initialize(...)` twice? In the old code, it looks like a single call to `$(...).sortable(...)` was sufficient.

Reply:
Well, it was initializing all items on the page in one call based on the class, and connecting all of them to each other! so I'd say this is a huge improvement. Let me check though, I might be able to pass it a list of the 2 IDs instead of calling it twice.

Reply:
Ah, I had completely missed the fact that the old selector actually matched multiple things on the page! To be clear, those questions were not necessarily asking for changes, just clarification - I would have been happy with the code as-is and the explanation you provided.